### PR TITLE
Ignore backport / autocut / dependabot branches for gradle checks on push

### PR DESCRIPTION
**Pull request by peterzhuamazon**
Signed-off-by: Peter Zhu <zhujiaxi@amazon.com>

### Description
Ignore backport / autocut / dependentbot branches for gradle checks.
These branches come and go and are not long-lived, thus no reason to even run gradle check on push.
They can run gradle check on pull_request_target event.
 
### Issues Resolved
https://github.com/opensearch-project/opensearch-build/issues/851
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/gradle-check.yml
Comment (peterzhuamazon) [reply .github/workflows/gradle-check.yml:4]: Hi,

@dreamer-89 you didnt get me these changes will be tested with PR event during PR but they dont need to be tested again when PUSH event happens.

See the description on top these branches get deleted soon after merge complete no reason to run the check again on a push, they dont need to upload codecov for a soon-to-delete branch.

Thanks.

Comment (dreamer-89) [reply .github/workflows/gradle-check.yml:4]: Thanks @peterzhuamazon for explaining this and apology for naivety here. Let me gather more info around lifecycle of these branches and get back. 

Adding more folks for review to unblock review.

Comment (reta) [reply .github/workflows/gradle-check.yml:4]: I think it makes sense: the only use of these branches is to create a pull requests to respective mainstream branch, no need to run any checks on pushes to any of those.

**Review by reta — APPROVED** [review]

**Review by owaiskazi19 — APPROVED** [review]

**Review by andrross — APPROVED** [review]
This looks good to me, but I have a related question...

We run the gradle check workflow when a PR is created against one of the mainstream branches, but we also run the check again when the merge is made? Is that right?

**Comment by peterzhuamazon** [comment]
> This looks good to me, but I have a related question...
> 
> We run the gradle check workflow when a PR is created against one of the mainstream branches, but we also run the check again when the merge is made? Is that right?

Exactly.

1. PR created/updated/reopened will trigger gradle check against the HEAD commit as part of pull_request_target event.
2. PR merged will trigger gradle check against the branch merged to as part of push event.

That is the point of this change so backport/depandabot/autocut will still trigger 1 and 2, while ignore the trigger towards those temp branches used to just create the PR.

**Comment by andrross** [comment]
Thanks @peterzhuamazon!

> PR merged will trigger gradle check against the branch merged to as part of push event.

I know this is a bit off topic, but I'm curious about exactly how this works. Does it happen before or after the push? What happens if it fails? If there's any documentation please let me know and I can RTFM :)

**Comment by peterzhuamazon** [comment]
> Thanks @peterzhuamazon!
> 
> > PR merged will trigger gradle check against the branch merged to as part of push event.
> 
> I know this is a bit off topic, but I'm curious about exactly how this works. Does it happen before or after the push? What happens if it fails? If there's any documentation please let me know and I can RTFM :)

Hi @andrross, 

if people creating a PR from their fork, due to the branch used to open PR is on fork, only PR event trigger gradle check, push wont.
Once PR merged due to the branch they merge to have a push happens, push event trigger.

if people creating a PR directly in this repo (aka dependabot, backport), due to those branches are created fresh, their 1st creation is considered a fresh push, thus trigger push. And since they also have PR thus trigger PR event. This means 2 gradle check and the push one is unnecessary.

There is literally a workflow in this repo to remove backport branches periodically therefore no need to even run those push events on these backport/dependabot branches.

Thanks.

**Comment by andrross** [comment]
I'm curious about the gradle check workflow that happens upon merge into main once the PR is approved. I see that the [gradle check failed](https://build.ci.opensearch.org/job/gradle-check/323/) for the workflow that ran after this PR was merged. How are these failures supposed to be handled? In this case it seems to be an existing flaky test (#3579) but if it was a problem caused by the merge is there any mechanism to handle it?

**Pull request merged**
